samples/go/server/server.go

```diff
@@ -0,0 +1,205 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"a2a/models"
+)
+
+// TaskHandler is a function type that handles task processing
+type TaskHandler func(task *models.Task, message *models.Message) (*models.Task, error)
+
+// A2AServer represents an A2A server instance
+type A2AServer struct {
+	handler     TaskHandler
+	port        int
+	basePath    string
+	taskStore   map[string]*models.Task
+	taskHistory map[string][]*models.Message
+	mu          sync.RWMutex
+}
+
+// NewA2AServer creates a new A2A server instance
+func NewA2AServer(handler TaskHandler, port int, basePath string) *A2AServer {
+	return &A2AServer{
+		handler:     handler,
+		port:        port,
+		basePath:    basePath,
+		taskStore:   make(map[string]*models.Task),
+		taskHistory: make(map[string][]*models.Message),
+	}
+}
+
+// Start starts the A2A server
+func (s *A2AServer) Start() error {
+	mux := http.NewServeMux()
+	mux.HandleFunc(s.basePath, s.handleRequest)
```

Comment:
This is an antipattern: basically, if want to use any other HTTP server, I would have a really hard time because you made the handler private...